### DISCUSSION TITLE
AxiosError: Request failed with status code 504

### DISCUSSION THREAD

**Discussion by rkoosaar** [firefly-iii/firefly-iii#9424] [Firefly III]
I'm running firefly III and data-importer on kubernetes and when Importing CSV files after a file I get the "AxiosError: Request failed with status code 504"

Does anyone know what might cause this? 
My FIREFLY_III_URL is internal cluster URL: "http://firefly.default.svc.cluster.local:8080"

**Comment by JC5** [comment]
A 504 is a gateway timeout. It looks like Firefly III can't be reached by the reverse proxy. Are all settings correct?

**Reply by rkoosaar** [reply]
My FIREFLY_III_URL is internal cluster URL: "http://firefly.default.svc.cluster.local:8080/" < it should not go via ingress-nginx as far as I understand.
It only gets 504 after processing a set of transactions almost like its getting too busy. 

Just thinking about the workflow:
- I open the "data-importer" URL (which is external via ingress-nginx)
- setup the csv import job
- start the job
after say 20 transactions I get 504 error. 
Looking at logs on the data-importer, it carries on importing the transactions its just the web front end that stops showing them. 
Which would indicate that the problem is probably still with ingress-nginx?
I have set these:
firefly-core:
"nginx.ingress.kubernetes.io/proxy-buffer-size: "16k""
data-importer:
"nginx.org/client-max-body-size: "0""
"nginx.org/proxy-buffer-size: "8k""

**Reply by JC5** [reply]
Depending on the setup it could timeout, although it's not a common occurrence. The data importer polls in the background using JS, that connection may be the one that's causing the timeouts? 

When in doubt, you could also use the command line options to import CSV files.

**Reply by rkoosaar** [reply]
I'm still looking into the nginx issue. Ill report back, probably would be a weekend job

**Reply by rkoosaar** [reply]
Still having this issue, but I'm largely ignoring it as its only the web UI side thats sad (in the background the import carries on).
One thing I noticed that if I ignore nginx-ingress and expose it directly then I dont have this issue. So it is for sure something nginx-ingress related. I've not figured out what bit of config I need to change to get past this. I've had a go at all the obvious things so far.

**Reply by JC5** [reply]
Mmmmm. It could be some hidden spam / server hammering thing? I wouldn't know otherwise :X

**Reply by KnightDoom** [reply]
Since you are using kubernetes, internal services do not go through ingresses (which are ways to expose services outside the cluster).
At worst, it may hit any namespace <-> namespace rules in your network manager (But since you are in default namespace, that wont be it)

The issue occurs during the mapping part of it, when DI is looking up your mapping to get the # for the account.
Most likely, a rate limit issue. wish we could stress test and set a rate limit (max #queries per sec)

**Comment by KnightDoom** [comment]
I have the same problem, a random 504 error which kinda "Freezes" the ui..

It looks like Firefly is being hammered during large import times, so 504s start being thrown (rate limit ?) (finally got around to importing my mint data 10000 transactions?)

regardless, As rkoosaar stated, Logging indicates that the process is still ongoing and eventually does "finish"..

now i wish i could get a report somewhere

**Reply by JC5** [reply]
https://docs.firefly-iii.org/references/faq/data-importer/import/#i-get-504-gateway-timeout-errors

**Comment by ssstain** [comment]
I've managed to fix the error before I stumbled on [another one](https://github.com/orgs/firefly-iii/discussions/8980)
 But this 504 type of an error can easily be fixed by adding (to nginx servers):
`fastcgi_read_timeout 3600;` in front of `fastcgi_pass...`
and
`proxy_read_timeout 3600;` in front of `proxy_pass...`
